packages/design-system-scripts/gulp/build.js

```diff
@@ -183,7 +182,8 @@ function compileJs(dir, options, changedPath) {
     })
     .then(() => {
       // If design system is using typescript, use tsc to generate definition files for tsx files
-      if (options.typescript) {
+      const unknownOrTypescriptPath = !changedPath || changedPath.match(/\.(ts|tsx)$/);
```

Comment:
Only generate type definitions if we changed a typescript file, or if we are rebuilding everything